sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ModuleIdentityLifecycleSamples.cs

```diff
@@ -120,34 +119,149 @@ public async Task<ModuleIdentity> CreateModuleIdentityAsync(DeviceIdentity devic
         {
-            return Task.CompletedTask;
-        }
+            SampleLogger.PrintHeader("LIST ALL MODULES");
 
-        public Task GetModuleIdentityAsync()
-        {
-            return Task.CompletedTask;
+            try
+            {
+                Console.WriteLine($"Listing all modules in device with Id: '{deviceIdentity.DeviceId}'\n");
+
+                Response<IReadOnlyList<ModuleIdentity>> response = await IoTHubServiceClient.Modules.GetIdentitiesAsync(deviceIdentity.DeviceId);
+
+                foreach (ModuleIdentity moduleIdentity in response.Value)
+                {
+                    SampleLogger.PrintSuccess($"\t- Module Id: '{moduleIdentity.ModuleId}', Device Id: '{moduleIdentity.DeviceId}', ETag: '{moduleIdentity.Etag}'");
+                }
+
+                return response.Value;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to list module identities due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task UpdateModuleIdentityAsync()
+        /// <summary>
+        /// Get a module identity
+        /// </summary>
+        /// <param name="deviceIdentity">The device identity module belongs to.</param>
+        /// <param name="moduleId">The unique identifier of the module to get.</param>
+        public async Task<ModuleIdentity> GetModuleIdentityAsync(DeviceIdentity deviceIdentity, string moduleId)
         {
-            return Task.CompletedTask;
-        }
+            SampleLogger.PrintHeader("GET A MODULE");
 
-        public Task GetModuleTwinAsync()
+            try
+            {
+                Console.WriteLine($"Getting module identity with Id: '{moduleId}'\n");
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.GetIdentityAsync(deviceIdentity.DeviceId, moduleId);
+
+                ModuleIdentity moduleIdentity = response.Value;
+
+                SampleLogger.PrintSuccess($"\t- Module Id: '{moduleIdentity.ModuleId}', Device Id: '{moduleIdentity.DeviceId}', ETag: '{moduleIdentity.Etag}'");
+
+                return moduleIdentity;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to get a module identity due to:\n{ex}");
+                throw;
+            }
+        }
+        
+        /// <summary>
+        /// Update a module identity
+        /// </summary>
+        /// <param name="moduleIdentity">Module identity to be updated.</param>
+        public async Task<ModuleIdentity> UpdateModuleIdentityAsync(ModuleIdentity moduleIdentity)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("UPDATE A MODULE");
+
+            try
+            {
+                Console.WriteLine($"Updating module identity with Id: '{moduleIdentity.ModuleId}'. Setting 'ManagedBy' property to: '{Environment.UserName}'\n");
```

Comment:
nit - we should log the current value of `moduleIdentity.ManagedBy` as well

Reply:
it's null, I don't think printing empty string is useful

Reply:
The idea here is to show the difference between original vs updated value. It doesn't really matter that the original is `null`, we want to highlight that the `IoTHubServiceClient.Modules.CreateOrUpdateIdentityAsync` operation did update the value.

What do you think?